test/integration/modules/screen-recorder.ts

```diff
@@ -0,0 +1,157 @@
+import path from 'path';
+import {exposeFunctionForTesting} from '../../../src/services/expose-functionality-module/expose-functionality.module';
+
+const media = {
+  video: {
+    tag: 'video',
+    type: 'video/x-matroska;codecs=avc1',
+    ext: '.mp4',
+  },
+};
+
+const LOCALSTORAGE_ISRECORDING = 'tests:screen-recorder:is-recording';
+const LOCALSTORAGE_FILEPATH = 'test:screen-recorder:file-path';
+
+export class ScreenRecorder {
+  private mediaRecorder: any;
+  private recordedBlobs: Array<Blob> = [];
+  private isRecording: boolean = false;
+  private filepath: string;
+
+  constructor() {
+    window.onbeforeunload = (): void => {
+      if (this.isRecording) {
+        this.stopRecordingAndSave();
+        window.localStorage.setItem(LOCALSTORAGE_ISRECORDING, JSON.stringify(true));
+      }
+    };
+
+    exposeFunctionForTesting('startRecording', (filepath: string) => {
+      this.startRecording(filepath);
+    });
+
+    exposeFunctionForTesting('cancelRecording', () => {
+      this.cancelRecording();
+    });
+
+    exposeFunctionForTesting('stopRecordingAndSave', () => {
+      this.stopRecordingAndSave();
+    });
+
+    this.isRecording = JSON.parse(window.localStorage.getItem(LOCALSTORAGE_ISRECORDING)) === true;
+
+    if (this.isRecording) {
+      this.filepath = JSON.parse(window.localStorage.getItem(LOCALSTORAGE_FILEPATH));
+      this.startRecording(this.filepath);
+    }
+  }
+
+  /**
+   * Records the BPMN Studio window
+   */
+  public async startRecording(filepath: string): Promise<void> {
+    this.filepath = filepath;
+    this.isRecording = true;
+    window.localStorage.setItem(LOCALSTORAGE_ISRECORDING, JSON.stringify(this.isRecording));
+
+    const bpmnStudioWindow = await this.getBpmnStudioWindow();
+
+    navigator.mediaDevices
+      .getUserMedia({
+        audio: false,
+        video: {
+          mandatory: {
+            chromeMediaSource: 'desktop',
+            chromeMediaSourceId: bpmnStudioWindow.id,
+            minWidth: 1280,
+            maxWidth: 1280,
+            minHeight: 720,
+            maxHeight: 720,
+          },
+        },
+      } as any)
+      .then(this.handleStream)
+      .catch(this.handleUserMediaError);
+  }
+
+  public async stopRecordingAndSave(): Promise<void> {
+    if (this.isRecording) {
+      window.localStorage.removeItem(LOCALSTORAGE_ISRECORDING);
+    }
+    this.isRecording = false;
+    this.mediaRecorder.stop();
+
+    const blob = new Blob(this.recordedBlobs, {type: media.video.type});
+    const arrayBuffer = await (blob as any).arrayBuffer();
+    const buffer = Buffer.from(arrayBuffer);
+
+    if (!fs.existsSync(path.dirname(this.filepath))) {
+      fs.mkdirSync(path.dirname(this.filepath), {recursive: true});
+    } else if (this.filepath.match(/-\d+.webm/)) {
+      const baseName = path.basename(this.filepath, '.webm');
+
+      const lastIndexOfMinus = baseName.lastIndexOf('-');
+      let number = JSON.parse(baseName.substr(lastIndexOfMinus + 1));
```

Comment:
You should improve the variable name since number does not tell anything about what its for.

Reply:
You could also use const here and add +1 at the end instead of using this line and `number++`

Reply:
Is there a reason why you use `JSON.parse` instead of `parseInt`?

Reply:
You could maybe change this to something like this:

```suggestion
      const videoIndex = parseInt(baseName.substr(lastIndexOfMinus + 1)) + 1;
```

Reply:
Das sollten zwei `const`s  sein: `currentVideoIndex` und `nextVideoIndex`